packages/cli-kit/src/public/node/base-command.test.ts

```diff
@@ -559,3 +559,176 @@ const deleteDefaultEnvironment = async (tmpDir: string): Promise<void> => {
   delete clone.environments.default
   await writeFile(joinPath(tmpDir, 'shopify.environments.toml'), encodeTOML({environments: clone} as any))
 }
+
+describe('removeDuplicatedPlugins', () => {
+  let capturedPlugins: Map<string, any> | undefined
+
+  class PluginTestCommand extends MockCommand {
+    async init() {
+      // Set up test plugins before calling super.init()
+      const initialPlugins = capturedPlugins
+      if (initialPlugins) {
+        this.config.plugins = new Map(initialPlugins)
+      }
+
+      const result = await super.init()
+
+      // Capture the plugins after init (which calls removeDuplicatedPlugins)
+      // eslint-disable-next-line require-atomic-updates
+      capturedPlugins = new Map(this.config.plugins)
+
+      return result
+    }
+  }
+
+  beforeEach(() => {
+    capturedPlugins = undefined
+  })
+
+  test('removes @shopify/app plugin when present', async () => {
+    await inTemporaryDirectory(async (tmpDir) => {
+      // Given - set up plugins to be injected
+      const mockPlugin1 = {name: '@shopify/app', version: '1.0.0'} as any
+      const mockPlugin2 = {name: '@shopify/plugin-ngrok', version: '1.0.0'} as any
+      const mockPlugin3 = {name: '@shopify/plugin-did-you-mean', version: '1.0.0'} as any
```

Comment:
Is there a reason not to remove extra plugins? It might not raise an error, but it seems like it could lead to non-obvious behavior.

Reply:
Extra plugins that don't conflict with the tunnel? Why would we want to remove them?

There are some devs working on custom plugins and also some generic ones, like did-you-mean, and they would stop working.